### DISCUSSION TITLE
Working on book site publishing single chapters at a time

### DISCUSSION THREAD

**Discussion by subsoap** [adityatelange/hugo-PaperMod#276] [Q&A]
https://discourse.gohugo.io/t/i-want-to-publish-chapters-of-multiple-books-looking-for-guidance/31454

I'm using this theme for this project.

One of the problems I'm facing is that if I have multiple branch bundle (what I'm trying to use to organize books) only the posts of a single one shows up on the main page. 

Using multiple branches other than posts also seems to break part of tags. The tag page populates, but when you click a tag for something in another branch bundle no posts/chapters which have that tag show up.

**Comment by subsoap** [comment]
Example project where the front page does not populate all posts of various sections, instead only picks one. Is this a bug of the theme or something else going on?

https://github.com/subsoap/hugo-book-project

**Reply by kdkasad** [reply]
This is expected. See [this Hugo wiki page](https://gohugo.io/functions/where/#mainsections) for more info.

TL;DR — set `mainSections` in `config.yml`:
```yaml
params:
    mainSections:
        - section1
        - section2
```

**Reply by subsoap** [reply]
Thank you, doing that does fix that but doing so changes the previous behavior of the next/previous buttons.

ATM I am stuck on getting the next/previous buttons to only select other pages of the same section.

I think I need to use either .CurrentSection or .Section here but I don't know how to properly yet.

```{{- $pages := where site.RegularPages "Type" "in" site.Params.mainSections }}```

**Reply by subsoap** [reply]
Got it I think 
```
{{- $pages := where site.RegularPages "Section" "==" .Section }}
```
I'm documenting process mostly here for now
https://discourse.gohugo.io/t/i-want-to-publish-chapters-of-multiple-books-looking-for-guidance/31454/13?u=subsoap
